.src/tsc.ts

```diff
@@ -0,0 +1,54 @@
+import * as ts from "npm:typescript";
+
+/**
+ * Compiles a TypeScript file to JavaScript using the TypeScript compiler API.
+ * 
+ * @param sourceFile - Path to the TypeScript source file
+ * @param outputDir - Directory where the compiled files should be written
+ */
+export async function compileTypeScript(
+  sourceFile: string,
+  outputDir: string,
+): Promise<void> {
+  // Use TypeScript compiler API programmatically
+  const compilerOptions: ts.CompilerOptions = {
+    target: ts.ScriptTarget.ES2015,
+    module: ts.ModuleKind.ES2015,
+    declaration: true,
```

Comment:
We now emit a declaration file instead of publishing *.ts files. This prevents errors for consumers that have `erasableSyntaxOnly` enabled (#116)